apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/AssessmentQuestionManualGrading.tsx

```diff
@@ -0,0 +1,205 @@
+import { QueryClient, useMutation } from '@tanstack/react-query';
+import { useState } from 'preact/compat';
+
+import type { AiGradingGeneralStats } from '../../../ee/lib/ai-grading/types.js';
+import { NuqsAdapter } from '../../../lib/client/nuqs.js';
+import type {
+  PageContextWithAuthzData,
+  StaffCourseInstanceContext,
+} from '../../../lib/client/page-context.js';
+import { QueryClientProviderDebug } from '../../../lib/client/tanstackQuery.js';
+import type { AssessmentQuestion, InstanceQuestionGroup } from '../../../lib/db-types.js';
+import type { RubricData } from '../../../lib/manualGrading.types.js';
+
+import { GroupInfoModal } from './GroupInfoModal.js';
+import type { InstanceQuestionRowWithAIGradingStats as InstanceQuestionRow } from './assessmentQuestion.types.js';
+import { AssessmentQuestionTable } from './assessmentQuestionTable.js';
+
+const queryClient = new QueryClient();
+
+export interface AssessmentQuestionManualGradingProps {
+  authzData: PageContextWithAuthzData['authz_data'];
+  course: StaffCourseInstanceContext['course'];
+  courseInstance: StaffCourseInstanceContext['course_instance'];
+  csrfToken: string;
+  instanceQuestions: InstanceQuestionRow[];
+  urlPrefix: string;
+  assessmentId: string;
+  assessmentQuestionId: string;
+  assessmentQuestion: AssessmentQuestion;
+  assessmentTid: string;
+  questionQid: string;
+  aiGradingMode: boolean;
+  groupWork: boolean;
+  rubricData: RubricData | null;
+  instanceQuestionGroups: InstanceQuestionGroup[];
+  courseStaff: { user_id: string; name: string | null; uid: string }[];
+  aiGradingStats: AiGradingGeneralStats | null;
+  numOpenInstances: number;
+  search: string;
+  isDevMode: boolean;
+}
+
+type AssessmentQuestionManualGradingInnerProps = Omit<
+  AssessmentQuestionManualGradingProps,
+  'search' | 'isDevMode'
+>;
+
+function AssessmentQuestionManualGradingInner({
+  authzData,
+  instanceQuestions,
+  course,
+  courseInstance,
+  urlPrefix,
+  csrfToken,
+  assessmentId,
+  assessmentQuestion,
+  assessmentTid,
+  questionQid,
+  aiGradingMode,
+  groupWork,
+  rubricData,
+  instanceQuestionGroups,
+  courseStaff,
+  aiGradingStats,
+  numOpenInstances,
+}: AssessmentQuestionManualGradingInnerProps) {
+  const [showSelectedModal, setShowSelectedModal] = useState(false);
+  const [showAllModal, setShowAllModal] = useState(false);
+  const [showUngroupedModal, setShowUngroupedModal] = useState(false);
```

Comment:
IMO these state names, and `handleShowSelectedModal` below, should have their names reflect that they're associated with grouping. In isolation, `showAllModal` is meaningless; `showGroupAllModal` is better.